Assets/_Game/Scripts/Inventory/InventoryObject.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using UnityEngine;
+using UnityEditor;
+
+[CreateAssetMenu(fileName = "New Inventory", menuName = "DLS/Inventory")]
+public class InventoryObject : ScriptableObject, ISerializationCallbackReceiver
+{
+    public string savePath = "PlayerInventory.save";
+    public List<InventorySlot> container = new List<InventorySlot>();
+    public int currentSize = 0;
+    public int maxSize = 8;
+    private ItemDatabaseObject database;
+
+    private void OnEnable()
+    {
+#if UNITY_EDITOR
+        database = (ItemDatabaseObject)AssetDatabase.LoadAssetAtPath<ItemDatabaseObject>("Assets/_Game/Resources/ItemDatabase.asset");
+#else
+        database = Resources.Load<ItemDatabaseObject>("ItemDatabase");
+#endif
+    }
+
+    public void AddItem(ItemObject _item, int _amount)
+    {
+        var foundItem = container.Find(x=> x.item == _item);
+        if (foundItem != null)
+        {
+            foundItem.amount += _amount;
+        }
+        else
+        {
+            if(currentSize < maxSize)
+            {
+                container.Add(new InventorySlot(_item, _amount));
+                currentSize++;
+            }
+        }
+        EventManager.InventoryChanged(this);
+
+    }
+
+    public void RemoveItem(ItemObject _item)
+    {
+        var foundItem = container.Find(x => x.item == _item);
+        if(foundItem != null)
+        {
+            if(foundItem.amount > 1)
+            {
+                foundItem.amount--;
+            }
+            else
+            {
+                container.Remove(foundItem);
+                currentSize--;
+            }
+        }
+        EventManager.InventoryChanged(this);
+    }
+
+    public ItemObject GetItem(string _name)
+    {
+        return container.Find(x=> x.item.name == _name).item;
+    }
+
+    public InventorySlot GetItemSlot(string _name)
+    {
+        return container.Find(x => x.item.name == _name);
+    }
+
+    public bool ItemExists(string _name)
+    {
+        return container.Exists(x=> x.item.name == _name);
+    }
+
+    public void OnBeforeSerialize()

```

Comment:
Empty Function

Reply:
No idea what to put in this function?

Reply:
Then remove it? If its an inbuilt Unity one like I'm pretty sure it is, there's a no-op already happening in the base class. No reason to keep an empty function around for no reason